controllers/secretgenerator/secretgenerator_controller_test.go

```diff
@@ -152,3 +155,94 @@ func TestExistingSecret(t *testing.T) {
 	g.Expect(generatedSecret.Labels).To(BeEmpty())
 	g.Expect(generatedSecret.StringData).To(BeEmpty())
 }
+
+func TestSecretNotFound(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+
+	secretName := "foo"
+	secretNs := "ns"
+
+	cli := newFakeClient()
+
+	r := secretgenerator.SecretGeneratorReconciler{
+		Client: cli,
+	}
+
+	_, err := r.Reconcile(ctx, reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      secretName,
+			Namespace: secretNs,
+		},
+	})
+	// secret not found, reconcile should end without error
+	g.Expect(err).ShouldNot(HaveOccurred())
+}
+
+func TestDeleteOAuthClientIfSecretNotFound(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+
+	secretName := "foo"
+	secretNs := "ns"
+
+	// secret expected to be deleted
+	existingSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: secretNs,
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+			Annotations: map[string]string{
+				annotations.SecretNameAnnotation: "bar",
+				annotations.SecretTypeAnnotation: "random",
+			},
+		},
+	}
+
+	// future left-over oauth client to be cleaned up during reconcile
+	existingOauthClient := oauthv1.OAuthClient{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "OAuthClient",
+			APIVersion: "oauth.openshift.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: secretNs,
+		},
+		Secret:       secretName,
+		RedirectURIs: []string{"https://foo.bar"},
+		GrantMethod:  oauthv1.GrantHandlerAuto,
+	}
+
+	cli := newFakeClient(&existingSecret, &existingOauthClient)
+
+	r := secretgenerator.SecretGeneratorReconciler{
+		Client: cli,
+	}
+
+	// delete secret
+	err := cli.Delete(ctx, &existingSecret)
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	// ensure the secret is deleted
+	err = cli.Get(ctx, client.ObjectKeyFromObject(&existingSecret), &existingSecret)
+	g.Expect(err).Should(HaveOccurred())
+	g.Expect(k8serr.IsNotFound(err)).Should(BeTrue())
+
+	_, err = r.Reconcile(ctx, reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      secretName,
+			Namespace: secretNs,
+		},
+	})
+	// secret not found, reconcile should clean-up left-over oauth client and end without error
+	g.Expect(err).ShouldNot(HaveOccurred())
+
+	// ensure the leftover OauthClient was deleted during reconcile
+	foundOauthClient := oauthv1.OAuthClient{}
+	err = cli.Get(ctx, client.ObjectKeyFromObject(&existingOauthClient), &foundOauthClient)
+	g.Expect(err).Should(HaveOccurred())
+	g.Expect(k8serr.IsNotFound(err)).Should(BeTrue())
```

Comment:
nit: gomega a some support to match errors so the expectation above can probably be re-written as `Expect(err).To(MatchError(k8sErr.IsNotFound))` (I have not tested it)

Reply:
guess line 231,246 `g.Expect(err).Should(HaveOccurred()) `not needed if we just wanna check the error is not found instance